kedro/framework/cli/starters.py

```diff
@@ -38,6 +43,81 @@
 }
 _STARTERS_REPO = "git+https://github.com/kedro-org/kedro-starters.git"
 
+
+@define(order=True)
+class KedroStarterSpec:
+    """Specification of custom kedro starter template.
+
+    Args:
+        name: name of the starter (name of the directory)
+        template_path: the path to git repository
+        directory: by default it will look for `name`, optionally override with custom directory.
+        origin: preserved field used by kedro internally to determine where does the starter
+        come from, user do not need to provide this field.
+
+    Example:
+    ::
+        >>> spec = KedroStarterSpec(
+                        name="astro-airflow-iris",
+                        template_path="git+https://github.com/kedro-org/kedro-starters.git",
+                        directory=None
+                                   )
+    """
+
+    name: str
+    template_path: str
+    directory: Optional[str] = None
+    origin: Optional[str] = field(init=False, default=None)
+
+    def __attrs_post_init__(self):
+        if not self.directory:
+            self.directory = self.name  # By default they are the same unless specified.
+
+    def format(self) -> Dict:
+        """Format a `KedroStarterSpec` to a nicely structured dictionary for printing.
+
+        Returns:
+            Dict: A formtted dictionary
+
+        Examples:
+        ::
+            >>> spec = KedroStarterSpec("astro-airflow-iris",
+                                    "git+https://github.com/kedro-org/kedro-starters.git")
+            >>> spec.format()
+                {
+                    "astro-airflow-iris": {
+                        "name": "astro-airflow-iris",
+                        "template_path": "git+https://github.com/kedro-org/kedro-starters.git",
+                                          }
```

Comment:
Related to the `directory` vs. `name` issue, I don't think this is quite the right format.